test_runner/regress/test_compatibility.py

```diff
@@ -220,9 +220,11 @@ def prepare_snapshot(
     for tenant in (repo_dir / "pgdatadirs" / "tenants").glob("*"):
         shutil.rmtree(tenant)
 
-    # Remove wal-redo temp directory
+    # Remove wal-redo temp directory if it is there
```

Comment:
```suggestion
    # Remove wal-redo temp directory if it exists. Newer pageserver versions don't create
    # them anymore, but old versions did.
```

Reply:
Done. Also added a similar code to the pageserver to clean up wal-redo directories automatically.